src/power_grid_model/_utils.py

```diff
@@ -749,3 +768,18 @@ def get_dataset_type(data: Dataset) -> DatasetType:
         raise ValueError("The dataset type could not be deduced because multiple dataset types match the data.")
 
     return next(iter(candidates))
+
+
+def get_comp_batch_size(comp_data: dict) -> int:
+    """
+    Get the batch size of the component update data.
+
+    Args:
+        comp_data: A dictionary representing the component data. The dictionary can be either columnar
+        or row-based.
+
+    Returns:
+        The length of the first value in the dictionary if the data is columnar, otherwise the length
+        of the dictionary itself.
+    """
+    return len(next(iter(comp_data.items()))[1]) if is_columnar(comp_data) else len(comp_data)
```

Comment:
This works only on uniform updates

Reply:
Then, I would argue that adding the specific types (not just `ComponentData`, as initially suggested) to the signature is a good idea, as it would inmediatly rule out non-uniform updates. Since there are different "branches" of data, batch, sparse, uniform, etc., I would argue that being specific will improve readability and reduce the posibility of mistakes in the future. Personally, I think we shouldn't use generics as hints if we can avoid to.

Reply:
Also nice to have to get values() instead of items() then [1]

Reply:
`ComponentData` won't work; not sized.